docs/lab/00.md

```diff
@@ -0,0 +1,532 @@
+
+:::
+
+As with any expression, we can construct a new instance of the structure as the last expression in the function body to implicitly return this new instance.
+
+```rust
+fn build_user(email: String, username: String) -> User {
+    User {
+        active: true,
+        username: username,
+        email: email,
+        sign_in_count: 1,
+    }
+}
+```
+#### Tuple structs
+
+Rust also supports structures that resemble tuples, called **tuple structs** . Tuple structures have the additional meaning provided by the structure name but do not have names associated with their fields; instead, they just have the field types. Tuple structures are useful when you want to name the entire tuple and make it a different type from other tuples, and when naming each field as in a regular structure would be wordy or redundant.
+
+```rust
+struct Color(i32, i32, i32);
+struct Point(i32, i32, i32);
+ 
+fn main() {
+    let black = Color(0, 0, 0);
+    let origin = Point(0, 0, 0);
+}
+```
+
+
+:::info 
+
+For a better understanding, please read [chapter 5](https://doc.rust-lang.org/book/ch05-00-structs.html) of the documentation.
+
+:::
+
+### String
+Rust has only one type of string in the core language, which is the string slice `str` which is usually seen in its borrowed form `&str`.
+
+The `String` type , which is provided by the Rust standard library rather than encoded in the main language, is a scalable, mutable, and owned UTF-8 encoded string type .
+
+#### Creating a new String
+```rust
+    let mut s = String::new();
+```
+This line creates a new empty string called `s`, which we can then load data into.
+
+We can use the `String::from` function or the `to_string` function to create a string from a string literal:
+```rust
+    let s = String::from("initial contents");
+```
+```rust
+let data = "initial contents";
+ 
+    let s = data.to_string();
+ 
+    // the method also works on a literal directly:
+    let s = "initial contents".to_string();
+```
+
+#### Adding to a string
+
+We can expand a string using the `push_str` method to add a string slice.
+```rust
+let mut s = String::from("foo");
+s.push_str("bar");
+```
+**The push** method takes **a single character** as a parameter and adds it to the string.
+
+```rust
+    let mut s = String::from("lo");
+    s.push('l');
+```
+#### Iteration Methods on Strings
+
+The best way to operate on pieces of strings is to be explicit about whether you want characters or bytes. For individual Unicode scalar values, use the `chars` method .
+```rust
+for c in "Зд".chars() {
+    println!("{c}");
+}
+```
+### Run the program
+In order to run the program we may be anywhere in the crate's folder and execute the command:
+
+```bash
+cargo run
+```
+
+## Exercises
+
+:::tip 
+
+If you don't have Rust installed, you can use [Rust Playground](https://play.rust-lang.org/) to solve the topics.
+
+:::
+
```

Comment:
I suggest asking students to go through some chapters of tour of rust before doing these exercises.

Reply:
@DTeodor-Alexaandru

Reply:
Wouldn't that make the laboratory too long? First 30min will be for organizational matters, then about 30min to go with them through the lab materials and after that they have to make the exercises. I wouldn t make the first lab that demanding:)))

Reply:
They might not be able to make the exercises without the tour.

Reply:
Proposal:
 - 30min organizational matters + take a look at the lab materials
 - 30min Tour of Rust
 - the rest for the exercises.
 
 Let's not forget we need time to grade them.

Reply:
Please write this in the lab, asking students to do some of the tour of rust tutorials.